code/modules/power/apc.dm

```diff
@@ -604,7 +604,8 @@
 		return
 	else if (istype(W, /obj/item/weapon/weldingtool) && opened && has_electronics==0 && !terminal)
 		var/obj/item/weapon/weldingtool/WT = W
-		if (WT.get_fuel() < 3)
+		if (!WT.isOn()) return
+		if (WT.get_fuel() < 3 && WT)

```

Comment:
`&& WT` should be checked for right after assignment. However, since it passes an istype check on line 605, it shouldn't be necessary.